### PR TITLE
Feature/closed api access

### PR DESCRIPTION
**Pull request by kss-espeo**

### CODE REVIEW

File: src/domain/common/usecase/DecryptUrlUseCase.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+class DecryptUrlUseCase {
+  private typeRegex = new RegExp(/(encrypted\(.*?\))/g);
+
+  constructor(private readonly privateKey: string) {
+    this.privateKey = privateKey;
+  }
+
+  async decrypt(url: string): Promise<string> {
+    let result = url;
```
Comment (przemyslawjanpietrzak) [inline src/domain/common/usecase/DecryptUrlUseCase.ts]: you don't have to, strings are immutable in js

File: src/domain/common/usecase/DecryptUrlUseCase.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    if (this.typeRegex.test(url)) {
```
Comment (przemyslawjanpietrzak) [inline src/domain/common/usecase/DecryptUrlUseCase.ts]: reverse if

File: src/domain/common/utils/error/InvalidContentTypeError.ts [hunk 3]
```diff
@@ -2,7 +2,7 @@ import { ErrorCode } from './ErrorCode';
 import InvalidRequestError from './InvalidRequestError';
 
 class InvalidContentTypeError extends InvalidRequestError {
-  constructor(message: string ) {
+  constructor(message: string) {
```
Comment (przemyslawjanpietrzak) [inline src/domain/common/utils/error/InvalidContentTypeError.ts]: ++

File: src/domain/common/usecase/DecryptUrlUseCase.ts (hunk 2 above)
Comment (kss-espeo) [reply src/domain/common/usecase/DecryptUrlUseCase.ts]: i'm not sure what do you mean. Can you elaborate, please?

File: src/domain/common/usecase/DecryptUrlUseCase.ts (hunk 1 above)
Comment (kss-espeo) [reply src/domain/common/usecase/DecryptUrlUseCase.ts]: What do you mean? `result` is supposed to be a modified version of `url`, in further processing I am recreating `result` based on its current value. I am not expecting to work on a reference of `url` there if that's what you mean (String is immutable in Java as well)

File: src/domain/common/usecase/DecryptUrlUseCase.ts (hunk 2 above)
Comment (kss-espeo) [reply src/domain/common/usecase/DecryptUrlUseCase.ts]: Agreed face2face.

File: src/domain/common/usecase/DecryptUrlUseCase.ts (hunk 1 above)
Comment (kss-espeo) [reply src/domain/common/usecase/DecryptUrlUseCase.ts]: Agreed face2face.

**Pull request merged**